frontend/src/components/modals/accountDelete/useDeleteAccountModal.tsx

```diff
@@ -1,11 +1,30 @@
+import { useMutation } from "@tanstack/react-query";
+import { IRequestError } from "helpers/error-interface";
+import { deleteUser } from "services/UsersService";
+import useToken from "hooks/useToken";
+import { FormEvent } from "react";
+import useFetchUser from "hooks/useFetchUser";
+
 export const useDeleteAccountModal = (hideModal: () => void) => {
-    const deleteAccountConfirm = () => {
-        //TODO: account delete logic
-        hideModal();
+    const { id } = useFetchUser();
+    const { setAccessToken } = useToken();
+
+    const mutation = useMutation<null, IRequestError>({
+        mutationFn: () => deleteUser(id!),
+        onSuccess: () => {
+            setAccessToken("");
```

Comment:
assing "null" to axios also here.

Reply:
What do you mean?

Reply:
call interceptior update here and assing 'null'